### PR TITLE
listener: undeprecate bind_to_port

### PR DESCRIPTION
**Pull request by tbarrella**
Commit Message:

listener: undeprecate bind_to_port

Signed-off-by: Taylor Barrella <tabarr@google.com>

Additional Description: See https://github.com/envoyproxy/envoy/issues/5355#issuecomment-747740929 and above for context
Risk Level: Low
Testing: Listener manager unit tests, grep
Docs Changes: Generated documentation for the proto field
Release Notes:
#5355
Deprecated: envoy_v3_api_field_config.listener.v3.Listener.deprecated_v1 (which was already hidden) was deprecated in favor of the new field envoy_v3_api_field_config.listener.v3.Listener.bind_to_port

### CODE REVIEW

**Comment by tbarrella** [comment]
(just a heads up that I plan to be taking a break from tomorrow to 1/4)

**Comment by tbarrella** [comment]
I removed the deprecation note because the field that was deprecated was hidden and named "deprecated_v1" anyway. I think the fact that it was hidden had caused the reference not to exist, breaking the docs build

**Comment by markdroth** [comment]
@htuch, I think you have context on this, so I'll defer to you for API approval on this particular PR.

File: source/server/listener_impl.cc [hunk 1]
```diff
@@ -73,6 +73,11 @@ bool usesProxyProto(const envoy::config::listener::v3::Listener& config) {
       config.filter_chains().empty() ? config.default_filter_chain() : config.filter_chains()[0],
       use_proxy_proto, false);
 }
+
+bool shouldBindToPort(const envoy::config::listener::v3::Listener& config) {
+  return !(!PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, bind_to_port, true) ||
```
Comment (htuch) [inline source/server/listener_impl.cc:78]: Can you apply De Morgan's law?

File: api/envoy/config/listener/v3/listener.proto [hunk 2]
```diff
@@ -272,4 +268,9 @@ message Listener {
   // The maximum length a tcp listener's pending connections queue can grow to. If no value is
   // provided net.core.somaxconn will be used on Linux and 128 otherwise.
   google.protobuf.UInt32Value tcp_backlog_size = 24;
+
+  // Whether the listener should bind to the port. A listener that doesn't
+  // bind can only receive connections redirected from other listeners that
+  // set use_original_dst parameter to true. Default is true.
```
Comment (htuch) [inline api/envoy/config/listener/v3/listener.proto:274]: Nit: add a `ref` link to the `use_original_dst` field.

**Review by htuch — CHANGES_REQUESTED** [review]
LGTM modulo nits.
/wait

**Review by htuch — DISMISSED** [review]
LGTM, thanks!

**Pull request merged**